src/plugins/robot-memory/event_trigger_manager.cpp

```diff
@@ -81,12 +81,16 @@ void EventTriggerManager::check_events()
 
   for(EventTrigger *trigger : triggers)
   {
-    while(trigger->oplog_cursor->more())
-    {
-      BSONObj change = trigger->oplog_cursor->next();
-      //logger_->log_info(name.c_str(), "Triggering: %s", change.toString().c_str());
-      //actually call the callback function
-      trigger->callback(change);
+    try {
```

Comment:
What are the expected errors we see here? I'm asking to understand why the try block is outside of the while loop. If it is a transient entry-specific recoverable error, it may make sense to put it in the inner loop.

Minor: the opening curly brace should be on the same line as the while, and there need to be spaces in front of the opening parentheses, e.g., "while (...) {",

Reply:
If I remember correctly, an exception was thrown when calling `trigger->oplog_cursor->more()`, which happens if the cursor is no longer valid.

Reply:
Ok, then it's the proper way.